c/include/ml-lxm-service-internal.h

```diff
@@ -0,0 +1,273 @@
+ *           std::cout.write(static_cast<const char *>(_raw), _size);
+ *           std::cout.flush();
+ *         }
+ *       }
+ *     }
+ *     break;
+ *   default:
+ *     break;
+ *   }
+ * }
+ * @endcode
+ */
+
+#ifndef __ML_LXM_SERVICE_INTERNAL_H__
+#define __ML_LXM_SERVICE_INTERNAL_H__
+
+#include <stdlib.h>
+#include <ml-api-service.h>
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/**
+ * @brief Enumeration for LXM service availability status.
+ */
+typedef enum
+{
+  ML_LXM_AVAILABILITY_AVAILABLE = 0,
+  ML_LXM_AVAILABILITY_DEVICE_NOT_ELIGIBLE,
+  ML_LXM_AVAILABILITY_SERVICE_DISABLED,
+  ML_LXM_AVAILABILITY_MODEL_NOT_READY,
+  ML_LXM_AVAILABILITY_UNKNOWN
+} ml_lxm_availability_e;
+
+/**
+ * @brief Checks LXM service availability.
+ * @param[out] status Current availability status.
+ * @return ML_ERROR_NONE on success, error code otherwise.
+ */
+int ml_lxm_check_availability (ml_lxm_availability_e * status);
+
+/**
+ * @brief A handle for lxm session.
+ */
+typedef void *ml_lxm_session_h;
+
+/**
+ * @brief Creates an LXM session.
+ * @param[out] session Session handle.
+ * @param[in] config_path Path to configuration file.
+ * @param[in] instructions Initial instructions (optional).
+ * @return ML_ERROR_NONE on success.
+ */
+int ml_lxm_session_create (ml_lxm_session_h * session, const char *config_path, const char *instructions);
+
+/**
+ * @brief Destroys an LXM session.
+ * @param[in] session Session handle.
+ * @return ML_ERROR_NONE on success.
+ */
+int ml_lxm_session_destroy (ml_lxm_session_h session);
+
+/**
+ * @brief A handle for lxm prompt.
+ */
+typedef void *ml_lxm_prompt_h;
+
+/**
+ * @brief Creates a prompt object.
+ * @param[out] prompt Prompt handle.
+ * @return ML_ERROR_NONE on success.
+ */
+int ml_lxm_prompt_create (ml_lxm_prompt_h * prompt);
+
+/**
+ * @brief Appends text to a prompt.
+ * @param[in] prompt Prompt handle.
+ * @param[in] text Text to append.
+ * @return ML_ERROR_NONE on success.
+ */
+int ml_lxm_prompt_append_text (ml_lxm_prompt_h prompt, const char *text);
+
+/**
+ * @brief Appends an instruction to a prompt.
+ * @param[in] prompt Prompt handle.
+ * @param[in] instruction Instruction to append.
+ * @return ML_ERROR_NONE on success.
+ */
+int ml_lxm_prompt_append_instruction (ml_lxm_prompt_h prompt, const char *instruction);
+
+/**
+ * @brief Destroys a prompt object.
+ * @param[in] prompt Prompt handle.
+ * @return ML_ERROR_NONE on success.
+ */
+int ml_lxm_prompt_destroy (ml_lxm_prompt_h prompt);
+
+/**
+ * @brief Sets runtime instructions for a session.
+ * @param[in] session Session handle.
+ * @param[in] instructions New instructions.
+ * @return ML_ERROR_NONE on success.
+ */
+int ml_lxm_session_set_instructions (ml_lxm_session_h session, const char *instructions);
+
+/**
+ * @brief Generation options for LXM responses.
+ */
+typedef struct
+{
+  double temperature; /* < Creativity control (0.0~2.0) */
+  size_t max_tokens;  /* < Maximum tokens to generate */
+} ml_lxm_generation_options_s;
```

Comment:
I suggest using ml-option handle for later work.
These options would be changed for various models.